src/coreclr/debug/daccess/request.cpp

```diff
@@ -2063,27 +2112,53 @@ ClrDataAccess::GetMethodTableTransparencyData(CLRDATA_ADDRESS mt, struct DacpMet
 }
 
 HRESULT
-ClrDataAccess::GetMethodTableForEEClass(CLRDATA_ADDRESS eeClass, CLRDATA_ADDRESS *value)
+ClrDataAccess::GetMethodTableForEEClass(CLRDATA_ADDRESS eeClassReallyCanonMT, CLRDATA_ADDRESS *value)
 {
-    if (eeClass == 0 || value == NULL)
+    if (eeClassReallyCanonMT == 0 || value == NULL)
         return E_INVALIDARG;
 
     SOSDacEnter();
-
-    PTR_EEClass pClass = PTR_EEClass(TO_TADDR(eeClass));
-    if (!DacValidateEEClass(pClass))
+    if (m_cdacSos != NULL)
     {
-        hr = E_INVALIDARG;
+        // Try the cDAC first - it will return E_NOTIMPL if it doesn't support this method yet. Fall back to the DAC.
+        hr = m_cdacSos->GetMethodTableForEEClass(eeClassReallyCanonMT, value);
+        if (FAILED(hr))
+        {
+            hr = GetMethodTableForEEClassImpl(eeClassReallyCanonMT, value);
+        }
+#ifdef _DEBUG
+        else
+        {
+            // Assert that the data is the same as what we get from the DAC.
+            CLRDATA_ADDRESS valueLocal;
+            HRESULT hrLocal = GetMethodTableForEEClassImpl(eeClassReallyCanonMT, &valueLocal);
+            _ASSERTE(hr == hrLocal);
+            _ASSERTE(*value == valueLocal);
+        }
+#endif
     }
     else
     {
-        *value = HOST_CDADDR(pClass->GetMethodTable());
+        hr = GetMethodTableForEEClassImpl (eeClassReallyCanonMT, value);
     }
-
     SOSDacLeave();
     return hr;
 }
 
+HRESULT
+ClrDataAccess::GetMethodTableForEEClassImpl(CLRDATA_ADDRESS eeClassReallyCanonMT, CLRDATA_ADDRESS *value)
+{
+    PTR_MethodTable pCanonMT = PTR_MethodTable(TO_TADDR(eeClassReallyCanonMT));
```

Comment:
Could you please change SOS to print parent method table here: https://github.com/dotnet/diagnostics/blob/main/src/SOS/Strike/strike.cpp#L1083C37-L1083C45 , and maybe delete places where it prints EEClass to avoid confusion.

Reply:
There is some discussion about the best way to do it in https://github.com/dotnet/diagnostics/pull/4751#discussion_r1649695256

Reply:
https://github.com/dotnet/diagnostics/pull/4758